src/HealthChecks/HealthChecks/src/DefaultHealthCheckService.cs

```diff
@@ -182,109 +182,82 @@ private static void ValidateRegistrations(IEnumerable<HealthCheckRegistration> r
 
         internal static class EventIds
         {
-            public static readonly EventId HealthCheckProcessingBegin = new EventId(100, "HealthCheckProcessingBegin");
-            public static readonly EventId HealthCheckProcessingEnd = new EventId(101, "HealthCheckProcessingEnd");
-
-            public static readonly EventId HealthCheckBegin = new EventId(102, "HealthCheckBegin");
-            public static readonly EventId HealthCheckEnd = new EventId(103, "HealthCheckEnd");
-            public static readonly EventId HealthCheckError = new EventId(104, "HealthCheckError");
-            public static readonly EventId HealthCheckData = new EventId(105, "HealthCheckData");
+            public const int HealthCheckProcessingBeginId = 100;
+            public const int HealthCheckProcessingEndId = 101;
+            public const int HealthCheckBeginId = 102;
+            public const int HealthCheckEndId = 103;
+            public const int HealthCheckErrorId = 104;
+            public const int HealthCheckDataId = 105;
+
+            public static readonly EventId HealthCheckProcessingBegin = new EventId(HealthCheckProcessingBeginId, nameof(HealthCheckProcessingBegin));
```

Comment:
Should these use `nameof`? If I can remember correctly the reason for using strings was that they wouldn't get renamed in a refactor if the field gets renamed and thus wouldn't cause breaking changes.

Ref: https://github.com/dotnet/aspnetcore/pull/11379#discussion_r295441810

Reply:
No it shouldn't use nameof. The EventId is part of the public API contract for log messages.

Reply:
These EventIds are actually no longer used in the library anymore with the source generator approach. They are only used by the unit tests. I left them here so they could re-use the `int` IDs, and to make them closer to the source of truth.

> If I can remember correctly the reason for using strings was that they wouldn't get renamed in a refactor if the field gets renamed and thus wouldn't cause breaking changes.

> The EventId is part of the public API contract for log messages.

With the source generator, the name of the partial method is used for the string ID portion of the EventId. Would we consider it "bad practice" to not explicitly list the `EventName` string in the attribute for this same reasoning?